home-application/src/main/java/com/softserveinc/ita/homeproject/application/exception/mapper/AlreadyExistExceptionMapper.java

```diff
@@ -0,0 +1,28 @@
+package com.softserveinc.ita.homeproject.application.exception.mapper;
+
+import com.softserveinc.ita.homeproject.homeservice.exception.AlreadyExistException;
+import com.softserveinc.ita.homeproject.model.ApiError;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * AlreadyExistExceptionMapper class is used to return exception
+ * response with HTTP status code, when the element already exists.
+ *
+ * @see javax.ws.rs.ext.ExceptionMapper
+ *
+ * @author Ihor Svyrydenko
+ */
+@Provider
+public class AlreadyExistExceptionMapper implements ExceptionMapper<AlreadyExistException> {
```

Comment:
Please introduce abstract BaseApiException. Extend `AlreadyExistException` and `NotFoundException` from it. Also pay attention to the naming of the exception, in this case, would be better to name it `NotFoundApiException`.

From the mapper's perspective please handle the mapping to the response in the abstract BaseApiExceptionMapper. Move the changeable data to the abstract methods.

Reply:
Ok, noted. Changes will be commited.